use-shopping-cart/src/@types/use-shopping-cart/index.d.ts

```diff
@@ -0,0 +1,132 @@
+declare module 'use-shopping-cart' {
+
+    export interface ProviderProps {
+        children: JSX.Element,
+        stripe: stripe.Stripe,
+        successUrl: string,
+        cancelUrl: string,
+        currency: string,
+        language: string,
+        billingAddressCollection: boolean,
+        allowedCountries: null | string[]
+    }
+
+    /**
+	 * Context provider to interact with Stripe API
+	 * @param children The Entry point to your app
+	 * @param stripe The stripe object
+	 * @param successUrl The redirect url for a successful sale
+	 * @param cancelUrl The redirect url for a cancelled sale
+	 * @param currency The preferred currency
+	 * @param language The language to be used in the checkout
+	 * @param billingAddressCollection Should the billing address be collected at the checkout. Defaults to false
+     * @param allowedCountries The allowed countries
+	 */
+    export const CartProvider: (props: ProviderProps) => JSX.Element;
+
+    export interface Product {
+        /** 
+         * The name of the product
+         */
+        name?: string;
+        /**
+         * The product sku
+         */
+        sku: string;
+        /**
+         * The price of the product
+         */
+        price: number;
+        /**
+         * An image of the product
+         */
+        image?: string;
+        /**
+         * The currency of the product
+         */
+        currency?: string;
+        /**
+         * Any additional properties
+         */
+        [propName: string]: any;
+    }
+
+    export type CartDetails = {
+        [sku: string]: Product
+    };
```

Comment:
I would suggest this as the solution for having CartDetails typed to include it's three unique values as well as all of Product's values.

```suggestion
    export interface CartEntry extends Product {
        /**
         * Amount of this product in the cart
         */
        quantity: number;
        /**
         * The total line item value, the price multiplied by the quantity
         */
        value: number;
        /**
         * Currency formatted version of value
         */
        formattedValue: string;
    }

    export interface CartDetails {
        [sku: string]: CartEntry
    };
```

Reply:
Solid suggestion!